src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -882,23 +882,8 @@ VarOrTerm Visitor::visitTypesafe(Parser::VarOrIriContext* ctx) {
 // ____________________________________________________________________________________
 GraphTerm Visitor::visitTypesafe(Parser::GraphTermContext* ctx) {
   if (ctx->numericLiteral()) {
-    auto literalAny = visitNumericLiteral(ctx->numericLiteral());
-    try {
-      auto intLiteral = literalAny.as<unsigned long long>();
-      return Literal{intLiteral};
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<long long>();
-      return Literal{intLiteral};
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<double>();
-      return Literal{intLiteral};
-    } catch (...) {
-    }
-    AD_CHECK(false);
+    auto wrapper = [](auto x) -> GraphTerm { return GraphTerm{Literal{x}}; };
+    return std::visit(wrapper, visitTypesafe(ctx->numericLiteral()));
```

Comment:
What about the following suggestion:
Implement a constructor for `Literal` directly from the return type of `numericLiteral()`.
Then move all the cases that start with `return Literal{...}` to the end and make them
`return visitAlternative<Literal>(...)`

Reply:
... This would really motivate switching to this c++23 fix for std::visit...